### PR TITLE
Respiration Enchantment and First Signs of Survival

### PR DESCRIPTION
**Pull request by aPunch**
Signed-off-by: aPunch theapunch@yahoo.com

To add the respiration enchantment, I needed drowning and suffocation to be implemented. Of course, the damage system will need to be expanded to account for things like armor, potion effects, etc., but for now this pull introduces some survival-like behavior for people to try out.

### CODE REVIEW

File: src/main/java/org/spout/vanilla/controller/VanillaActionController.java [hunk 1]
```diff
@@ -298,26 +301,15 @@ public void damage(int amount, DamageCause cause) {
...
 	 * @param cause cause of this controller being damaged
 	 * @param damager controller that damaged this controller
-	 * @param sendHurtMessage whether or not to send a hurt message
 	 */
-	public void damage(int amount, DamageCause cause, VanillaActionController damager, boolean sendHurtMessage) {
+	public void damage(int amount, DamageCause cause, VanillaActionController damager) {
 		setHealth(getHealth() - amount, new HealthChangeReason(HealthChangeReason.Type.DAMAGE));
 		lastDamager = damager;
-		if (sendHurtMessage) {
+		if (amount > 0) {
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/controller/VanillaActionController.java]: I believe this message still gets sent even if the player is not damaged. Take snowballs for example.

Comment (aPunch) [inline src/main/java/org/spout/vanilla/controller/VanillaActionController.java]: I've always thought that playing the hurt animation for zero damage was a bug or an oversight. :P

Comment (windy1) [inline src/main/java/org/spout/vanilla/controller/VanillaActionController.java]: How would the opponent know if the snowball hit or not? :p

File: src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java [hunk 2]
```diff
@@ -81,7 +84,8 @@
 	protected Window activeWindow = new DefaultWindow(this);
 	protected String tabListName;
 	protected GameMode gameMode;
-	protected int distanceMoved, miningDamagePeriod = VanillaConfiguration.PLAYER_SPEEDMINING_PREVENTION_PERIOD.getInt(), miningDamageAllowance = VanillaConfiguration.PLAYER_SPEEDMINING_PREVENTION_ALLOWANCE.getInt();
+	protected int distanceMoved, miningDamagePeriod = VanillaConfiguration.PLAYER_SPEEDMINING_PREVENTION_PERIOD.getInt(),
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java:87]: Line break :/

**Pull request merged**